include/boost/geometry/algorithms/detail/relate/linear_areal.hpp

```diff
@@ -1299,9 +1289,21 @@ struct linear_areal
 
         // calculate inside or outside based on side_calc
         // this is simplified version of a check from equal<>
-        template <typename SideCalc>
-        static inline bool calculate_from_inside_sides(SideCalc const& side_calc)
+        template <typename Strategy,
+                  typename Pi, typename Pj, typename Pk,
+                  typename Qi, typename Qj, typename Qk>
+        static inline
+        bool calculate_from_inside_sides(Pi const& pi, Pj const& pj, Pk const& pk,
+                                         Qi const& qi, Qj const& qj, Qk const& qk,
+                                         Strategy const& strategy)
         {
+            la_side_calculator
+            <
+                Strategy,
+                Pi, Pj, Pk,
+                Qi, Qj, Qk
+            > side_calc(pi, pj, pk, qi, qj, qk, strategy);
```

Comment:
If that's the preferred way then I suggest to remove `la_side_calculator` entirely and replace function calls like `pk_wrt_p()` with corresponding strategy calls.

Reply:
I don't want to make the PR bigger. I don't say it's the preferred way - it is a more convenient way than before, because it does not need conversions.

Reply:
Ok got it.